airbyte-api/src/main/openapi/config.yaml

```diff
@@ -2502,6 +2551,53 @@ components:
         - auto
         - us
         - eu
+    # BuilderProject
+    ConnectorBuilderProjectId:
+      type: string
+      format: uuid
+    ConnectorBuilderProjectDetails:
+      type: object
+      required:
+        - name
+      properties:
+        name:
+          type: string
+        draftManifest:
+          type: object
+          additionalProperties: true
+    ConnectorBuilderProjectIdWithWorkspaceId:
+      type: object
+      required:
+        - workspaceId
+        - builderProjectId
+      properties:
+        workspaceId:
+          $ref: "#/components/schemas/WorkspaceId"
+        builderProjectId:
+          $ref: "#/components/schemas/ConnectorBuilderProjectId"
+    ExistingConnectorBuilderProjectWithWorkspaceId:
+      type: object
+      required:
+        - workspaceId
+        - builderProjectId
+        - builderProject
+      properties:
+        workspaceId:
+          $ref: "#/components/schemas/WorkspaceId"
+        builderProjectId:
+          $ref: "#/components/schemas/ConnectorBuilderProjectId"
+        builderProject:
+          $ref: "#/components/schemas/ConnectorBuilderProjectDetails"
+    ConnectorBuilderProjectWithWorkspaceId:
+      type: object
+      required:
+        - workspaceId
+        - builderProject
+      properties:
+        workspaceId:
+          $ref: "#/components/schemas/WorkspaceId"
+        builderProject:
+          $ref: "#/components/schemas/ConnectorBuilderProjectDetails"
```

Comment:
@flash1293 I don't think this should be pointing to `ConnectorBuilderProjectDetails`, because that struct has `builderProjectId` as a required field. 

This `ConnectorBuilderProjectWithWorkspaceId` struct is used for the request body `/connector_builder_projects/create`, so when that is called the webapp will not know what the `builderProjectId` should be, however with the current API setup this field is required when calling it.

So I think we'll need a new struct similar to `ConnectorBuilderProjectDetails` but with `builderProjectId` not being required, to be used here

Reply:
You are right, we need different classes for the request and the response. However, as `ConnectorBuilderProjectDetails` is only used for writing on this PR I will fix this issue on https://github.com/airbytehq/airbyte-platform/pull/46